staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -372,3 +342,55 @@ func scaleFromCustomResource(cr *unstructured.Unstructured, specReplicasPath, st
 
 	return scale, foundSpecReplicas, nil
 }
+
+type scaleUpdatedObjectInfo struct {
+	reqObjInfo         rest.UpdatedObjectInfo
+	specReplicasPath   string
+	statusReplicasPath string
+	labelSelectorPath  string
+}
+
+func (i *scaleUpdatedObjectInfo) Preconditions() *metav1.Preconditions {
+	return nil
+}
+
+func (i *scaleUpdatedObjectInfo) UpdatedObject(ctx context.Context, oldObj runtime.Object) (runtime.Object, error) {
+	cr := oldObj.DeepCopyObject().(*unstructured.Unstructured)
```

Comment:
why the deep copy?

Reply:
I think the original object should not be modified and decide to make a copy.

Reply:
And I found that if we dont deep copy here, the server would not return a conflict error even if resourceVersion mismatched.

Reply:
I don't know why and I'll try to figure it out

Reply:
no, the deepcopy makes sense. I see the default impl does the same thing